hud/src/androidLibMain/kotlin/AndroidHUD.kt

```diff
@@ -97,17 +121,53 @@ class AndroidHUD private constructor(viewResId: Int, hudConfig: HudConfig, priva
         }
     }
 
+    private sealed class DialogState {
+        object Gone: DialogState()
+        object Visible: DialogState()
+    }
+
     private val loadingDialog = LoadingDialog.newInstance(viewResId, hudConfig)
+    private var dialogState = MutableLiveData<DialogState>()
+
+    init {
+        checkNotNull(uiContextObserver.uiContextData) { "Please subscribe builder to existing UI context before building HUD" }
```

Comment:
This might not strictly be needed?

Reply:
It's still possible to call present hud without subscribing and it will does nothing.
This check will explain why it doesn't work.

Reply:
We can make it work so that it will work the moment the subscribe is done, the HUS is showing, after all HUD is just a state now (e.g. in the view model)

But this can also be done in the next iteration.

Reply:
Yes, I was thinking about it. But for the end user it will be still confusing because it does nothing on present call, but later. Anyway we can discuss it for the future

Reply:
When we add
```kotlin
HUD.showWhile { // not the final name 😆
   // 
}
```
I guess it does not matter as much. Just have to make sure internally it can handle this from a point where `Lifecycle`/`FragmentManager` are not set yet.